include/circt/Dialect/FIRRTL/FIRRTLIntrinsics.h

```diff
@@ -40,7 +40,7 @@ class IntrinsicConverter {
   virtual bool check() { return false; }
 
   /// Transform an instance of the intrinsic.
-  virtual void convert(InstanceOp op) = 0;
+  virtual LogicalResult convert(InstanceOp op) = 0;
```

Comment:
What's the utility of `check` if `convert` can fail? Some sense that it should have worked but for un-checked (unreasonable to check up front?) reasons it was unable to be converted?

Reply:
Ah, convert takes the instance specifically which might need to fail because of the instance (like if it has annotations, as is checked here :upside_down_face: ), okay :+1: .

Reply:
Yeah exactly! :-) The EICG thingy is the first thing that can actually fail in this way. Although, if we decide to reject annotations on intrinsics in general, we can get rid of the EICG-specific check again.